core/services/runs.go

```diff
@@ -368,17 +374,25 @@ func meetsMinimumConfirmations(
 	run *models.JobRun,
 	taskRun *models.TaskRun,
 	currentHeight *models.Big) bool {
-	if run.CreationHeight == nil || currentHeight == nil {
+	if !taskRun.MinimumConfirmations.Valid || run.CreationHeight == nil || currentHeight == nil {
 		return true
 	}
 
 	diff := blockNumberDifference(currentHeight, run.CreationHeight)
-	return diff >= taskRun.MinimumConfirmations
+	return diff >= taskRun.MinimumConfirmations.Uint32
 }
 
-func blockNumberDifference(currentHeight, creationHeight *models.Big) uint64 {
-	diff := new(big.Int).Sub(currentHeight.ToInt(), creationHeight.ToInt())
-	return diff.Uint64() + 1 // creation of runlog alone warrants 1 confirmation
+func blockNumberDifference(currentHeight, creationHeight *models.Big) uint32 {
+	bigDiff := new(big.Int).Sub(currentHeight.ToInt(), creationHeight.ToInt())
+	diff := bigDiff.Uint64() + 1 // creation of runlog alone warrants 1 confirmation
+
+	small := uint32(diff)
+	if uint64(small) != diff {
```

Comment:
It's really late to be doing this check, how can we push this error closer to the user?

Reply:
I'm all for that idea, I'm not sure how to pull it off. This doesn't respond to a user event, but rather a `newHead` confirmation in `ResumeConfirmingTask`, which is internal to the job runner and not in response to a user setting or event.

Reply:
If `confirmations` for task is "past max uint32" then isn't that the same as saying the minimum number of confirmations is met?

Reply:
Yes you are correct. As it stands now, this calculation is between arbitrary block heights which then gets compared  to confirmations. This is it should be safe to return `math.MaxUint32` in this private helper method:

```
	diff := blockNumberDifference(currentHeight, run.CreationHeight)
	return diff >= taskRun.MinimumConfirmations.Uint32
```

Should we just have better method names and documentation, and remove the warning altogether?

Reply:
I've refactored the code and removed the warning to better reflect your observation.

Reply:
Per discussion offline, comparison has been changed to promote lower precision values.